src/cmdline/RootCommands.cpp

```diff
@@ -0,0 +1,287 @@
+static bool   _verbose         = false;
+static bool   _headless        = false;
+#ifndef DISABLE_NETWORK
+static uint32 _port            = 0;
+#endif
+static utf8 * _userDataPath    = nullptr;
+static utf8 * _openrctDataPath = nullptr;
+
+static const CommandLineOptionDefinition StandardOptions[]
+{
+    { CMDLINE_TYPE_SWITCH,  &_help,            'h', "help",              "show this help message and exit"                            },
+    { CMDLINE_TYPE_SWITCH,  &_version,         'v', "version",           "show version information and exit"                          },
+    { CMDLINE_TYPE_SWITCH,  &_noInstall,       'n', "no-install",        "do not install scenario if passed"                          },
+    { CMDLINE_TYPE_SWITCH,  &_all,             'a', "all",               "show help for all commands"                                 },
+    { CMDLINE_TYPE_SWITCH,  &_about,           NAC, "about",             "show information about " OPENRCT2_NAME                      },
+    { CMDLINE_TYPE_SWITCH,  &_verbose,         NAC, "verbose",           "log verbose messages"                                       },
+    { CMDLINE_TYPE_SWITCH,  &_headless,        NAC, "headless",          "run " OPENRCT2_NAME " headless"                             },
+#ifndef DISABLE_NETWORK
+    { CMDLINE_TYPE_INTEGER, &_port,            NAC, "port",              "port to use for hosting or joining a server"                },
+#endif
+    { CMDLINE_TYPE_STRING,  &_userDataPath,    NAC, "user-data-path",    "path to the user data directory (containing config.ini)"    },
+    { CMDLINE_TYPE_STRING,  &_openrctDataPath, NAC, "openrct-data-path", "path to the OpenRCT2 data directory (containing languages)" },
+    OptionTableEnd
+};
+
+static exitcode_t HandleNoCommand(CommandLineArgEnumerator * enumerator);
+static exitcode_t HandleCommandEdit(CommandLineArgEnumerator * enumerator);
+static exitcode_t HandleCommandIntro(CommandLineArgEnumerator * enumerator);
+static exitcode_t HandleCommandHost(CommandLineArgEnumerator * enumerator);
+static exitcode_t HandleCommandJoin(CommandLineArgEnumerator * enumerator);
+
+static void PrintAbout();
+static void PrintVersion();
+static void PrintLaunchInformation();
+
+const CommandLineCommand CommandLine::RootCommands[]
+{
+    // Main commands
+    DefineCommand("",      "<uri>",      StandardOptions, HandleNoCommand   ),
+    DefineCommand("edit",  "<uri>",      StandardOptions, HandleCommandEdit ),
+    DefineCommand("intro", "",           StandardOptions, HandleCommandIntro),
+#ifndef DISABLE_NETWORK
+    DefineCommand("host",  "<uri>",      StandardOptions, HandleCommandHost ),
+    DefineCommand("join",  "<hostname>", StandardOptions, HandleCommandJoin ),
+#endif
+
+    // Sub-commands
+    DefineSubCommand("screenshot", CommandLine::ScreenshotCommands),
+    DefineSubCommand("sprite",     CommandLine::SpriteCommands    ),
+
+    CommandTableEnd
+};
+
+static const CommandLineExample CommandLine::RootExamples[]
```

Comment:
"Static declaration of 'RootExamples' follows non-static declaration"  Xcode doesn't like this either.